### ISSUE TITLE
Rnw-mode "loses" the effect of its file-local variables when exiting R-mode

### ISSUE THREAD

**Issue by EmmanuelCharpentier** [emacs-ess/ESS#487]
Note : this is a **duplicate** of an issue posted a few days ago on the `ess-bugs list`, not yet acknowledged nor included in the mailing list archive.

I noticed the same issue when trying to use Rnw as the host mode pf a polymode, and Vitalie Spinu, who couldn't reproduce my problem, suggested to post my findings abot `Rnw-mode` here.

How to reproduce : use the `initMinimal.el` fle as the init filen to edit `Essai0.el` : `emacs -q -l initMinimal.el Essai0` : 

[MinimalExample.zip](https://github.com/emacs-ess/ESS/files/1717133/MinimalExample.zip)

* After a fresh start, the file variable `TeX-master` is set to `"Essai0"`(as seen by `M-: TeX-master`), thus allowing AUCTeX's `C-c C-c` to compile `Essai0.Rnw`. At this point : 
    - the mode indicator of emacs reads `(LaTeX/PS Ref Wrap Noweb NWFL)`, 
    - `M-: file-local-variables-alist` returns `((TeX-master . "Essai0") (TeX-engine . default) (mode . Rnw-mode))`.
* Move after `2+2` in `Firstrchunk`. At this point : 
    - the Emacs mode indicator reads `(ESS[S] [none] ElDoc Noweb NWFL)`, 
    - `M-: TeX-master` returns `t`,
    - `M-: file-local-variables-alist` still returns `((TeX-master . "Essai0") (TeX-engine . default) (mode . Rnw-mode))`.
* Move _outside* the R chunk (e. g. on the empty line below it). At this point : 
    - The Emacs mode line reads `LaTeX/PS Ref Wrap Noweb NWFL)`
    - `M-: TeX-master` still returns `t`,
    - `M-: file-local-variables-alist` still returns `((TeX-master . "Essai0") (TeX-engine . default) (mode . Rnw-mode))`.

In other words, `TeX-master` is always in the file's local variables, but is _ignored_ or _unset_ after passing in an Rnw-mode chunk. Ditto for TeX-engine, BTW.

The consequences are what one can expect : `M-r` correctly compiles `Essai0.Rnw` to `Essai0.tex`, but `C-c C-c` tries to LaTeX `Essai0.Rnw`, which typesets the R chunk literally, and not its outputs. Further trials show that the value of `TeX-engine` is also ignored, and that after moving out of an R chunk, `C-c C-c` will invoke `pdflatex` (the  default) and _not_ `xelatex` nor `lualatex` if so requited in the file variables.

All is fixed by `C-u C-c C-n` (reset AUXTeX). But it is annoying...

I'll try to debug that, but I'm a bit out of my depth here...
